lib/omniauth/strategies/digitalocean.rb

```diff
@@ -8,46 +7,37 @@ module Strategies
     # Usage:
     #    use OmniAuth::Strategies::Digitalocean, 'consumerkey', 'consumersecret', :scope => 'read write', :display => 'plain'
     #
-    class Digitalocean < OmniAuth::Strategies::OAuth2
+    class DigitalOcean < OmniAuth::Strategies::OAuth2
       AUTHENTICATION_PARAMETERS = %w(display state scope)
-      BASE_URL = "https://cloud.digitalocean.com"
 
-      option :name, :digitalocean
+      # Name of this strategy
+      option :name, 'digitalocean'
 
-      unless OmniAuth.config.test_mode
-        option :client_options, {
-          :authorize_url => "#{BASE_URL}/v1/oauth/authorize",
-          :token_url => "#{BASE_URL}/v1/oauth/token",
-          :site => BASE_URL
-        }
-      else
-        option :client_options, {
-          :authorize_url => "http://localhost:3000/v1/oauth/authorize",
-          :token_url => "http://localhost:3000/v1/oauth/token",
-          :site => "http://localhost:3000"
-        }
-      end
+      # Client options ( see https://github.com/intridea/omniauth-oauth2 )
+      option :client_options, {
+        site: 'https://api.digitalocean.com',
+        authorize_url: 'https://cloud.digitalocean.com/v1/oauth/authorize',
+        token_url: 'https://cloud.digitalocean.com/v1/oauth/token',
+      }
 
       option :authorize_options, AUTHENTICATION_PARAMETERS
 
-      uid do
-        access_token.params['uid']
-      end
+      uid { raw_info['uuid'] }
 
       info do
-        access_token.params['info']
-      end
-
-      # Hook useful for appending parameters into the auth url before sending
-      # to provider.
-      def request_phase
-        super
+        {
+          droplet_limit: raw_info['droplet_limit'],
+          email: raw_info['email'],
+          uuid: raw_info['uuid'],
+          email_verified: raw_info['email_verified'],
+        }
       end
 
-      # Hook used after response with code from provider. Used to prep token
-      # request from provider.
-      def callback_phase
-        super
+      ##
+      # Retrieves the information about the user
+      #
+      def raw_info
+        @raw_info ||= access_token.get('v2/account').parsed['account']
```

Comment:
`raw_info` should be pulling from the `info` hash in the raw response, it's great to hit the external api but we should do that in the `user_info` method so that it can be turned off if folks don't want the extra API call: http://www.rubydoc.info/gems/omniauth/OmniAuth%2FStrategy%3Askip_info%3F